### ISSUE TITLE
Uncaught exceptions in handlers result in messy TBs in logs during test suite runs

### ISSUE THREAD

**Issue by acrellin** [skyportal/skyportal#827]
```
[!] Error in `/api/comment/187`: Insufficient permissions.
WARNING:tornado.access:400 GET /api/comment/187 (127.0.0.1) 4.82ms
ERROR:tornado.application:Uncaught exception GET /api/comment/199 (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='GET', uri='/api/comment/199', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 30, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/skyportal/handlers/api/comment.py", line 30, in get
    comment = Comment.get_if_owned_by(comment_id, self.current_user)
  File "/home/stefan/src/skyportal/skyportal/models.py", line 1258, in get_if_owned_by
    comment.author_info = comment.construct_author_info_dict()
AttributeError: 'NoneType' object has no attribute 'construct_author_info_dict'
[!] Error in `/api/comment/199`: 'NoneType' object has no attribute 'construct_author_info_dict'
```

```
ERROR:tornado.application:Uncaught exception POST /api/photometry (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='POST', uri='/api/photometry', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 30, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 51, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/skyportal/handlers/api/photometry.py", line 151, in post
    data = self.get_json()
  File "/home/stefan/src/skyportal/baselayer/app/handlers/base.py", line 112, in get_json
    raise Exception('Please ensure posted data is of type application/json')
Exception: Please ensure posted data is of type application/json
[!] Error in `/api/photometry`: Please ensure posted data is of type application/json
```

```
ERROR:tornado.application:Uncaught exception POST /api/photometry (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='POST', uri='/api/photometry', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 30, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 51, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/skyportal/handlers/api/photometry.py", line 365, in post
    raise ValidationError(
marshmallow.exceptions.ValidationError: Instrument ZTF_efcdce12-3dfd-4bef-a527-a532cb56cf76 has no filter bessellv.
```

```
ERROR:tornado.application:Uncaught exception POST /api/user (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='POST', uri='/api/user', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy_utils/types/phone_number.py", line 88, in __init__
    self._phone_number = phonenumbers.parse(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/phonenumbers/phonenumberutil.py", line 2799, in parse
    raise NumberParseException(NumberParseException.NOT_A_NUMBER,
phonenumbers.phonenumberutil.NumberParseException: (1) The string supplied did not seem to be a phone number.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 30, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 51, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/skyportal/handlers/api/user.py", line 169, in post
    user_id = add_user_and_setup_groups(
  File "/home/stefan/src/skyportal/skyportal/handlers/api/user.py", line 29, in add_user_and_setup_groups
    DBSession().flush()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2459, in flush
    self._flush(objects)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2597, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2557, in _flush
    flush_context.execute()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 586, in execute
    persistence.save_obj(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 239, in save_obj
    _emit_insert_statements(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 1137, in _emit_insert_statements
    result = cached_connections[connection].execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 287, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1101, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1181, in _execute_context
    self._handle_dbapi_exception(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1179, in _execute_context
    context = constructor(dialect, self, conn, *args)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 738, in _init_compiled
    param = dict(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 741, in <genexpr>
    processors[key](compiled_params[key])
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 1205, in process
    return impl_processor(process_param(value, dialect))
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy_utils/types/phone_number.py", line 183, in process_bind_param
    value = PhoneNumber(value, region=self.region)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy_utils/types/phone_number.py", line 100, in __init__
    raise PhoneNumberParseException(
sqlalchemy_utils.types.phone_number.PhoneNumberParseException: (1) (1) The string supplied did not seem to be a phone number.
```

**Comment by stefanv** [comment]
```
ERROR:tornado.application:Uncaught exception PATCH /api/filters/17 (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='PATCH', uri='/api/filters/17', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1243, in _execute_context
    self.dialect.do_execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.ForeignKeyViolation: insert or update on table "filters" violates foreign key constraint "filters_group_id_fkey"
DETAIL:  Key (group_id)=(0) is not present in table "groups".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 30, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/skyportal/handlers/api/filter.py", line 198, in patch
    DBSession().commit()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1027, in commit
    self.transaction.commit()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 494, in commit
    self._prepare_impl()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 473, in _prepare_impl
    self.session.flush()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2459, in flush
    self._flush(objects)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2597, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2557, in _flush
    flush_context.execute()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 586, in execute
    persistence.save_obj(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 230, in save_obj
    _emit_update_statements(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 995, in _emit_update_statements
    c = cached_connections[connection].execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 287, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1101, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1247, in _execute_context
    self._handle_dbapi_exception(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1243, in _execute_context
    self.dialect.do_execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "filters" violates foreign key constraint "filters_group_id_fkey"
DETAIL:  Key (group_id)=(0) is not present in table "groups".

[SQL: UPDATE filters SET modified=timezone(%(timezone_1)s, CURRENT_TIMESTAMP), group_id=%(group_id)s WHERE filters.id = %(filters_id)s]
[parameters: {'timezone_1': 'UTC', 'group_id': 0, 'filters_id': 17}]
(Background on this error at: http://sqlalche.me/e/gkpj)
[!] Error in `/api/filters/17`: (psycopg2.errors.ForeignKeyViolation) insert or update on table "filters" violates foreign key constraint "filters_group_id_fkey"
DETAIL:  Key (group_id)=(0) is not present in table "groups".

[SQL: UPDATE filters SET modified=timezone(%(timezone_1)s, CURRENT_TIMESTAMP), group_id=%(group_id)s WHERE filters.id = %(filters_id)s]
[parameters: {'timezone_1': 'UTC', 'group_id': 0, 'filters_id': 17}]
(Background on this error at: http://sqlalche.me/e/gkpj)
```

**Comment by stefanv** [comment]
```
ERROR:tornado.application:Uncaught exception GET /api/sources/%7Bpublic_source.id%7D (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:5000', method='GET', uri='/api/sources/%7Bpublic_source.id%7D', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1243, in _execute_context
    self.dialect.do_execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.ForeignKeyViolation: insert or update on table "sourceviews" violates foreign key constraint "sourceviews_obj_id_fkey"
DETAIL:  Key (obj_id)=({public_source.id}) is not present in table "objs".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/stefan/src/skyportal/baselayer/app/access.py", line 30, in wrapper
    return method(self, *args, **kwargs)
  File "/home/stefan/src/skyportal/skyportal/handlers/api/source.py", line 182, in get
    register_source_view(
  File "/home/stefan/src/skyportal/skyportal/handlers/api/internal/source_views.py", line 74, in register_source_view
    DBSession.commit()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/scoping.py", line 162, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 1027, in commit
    self.transaction.commit()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 494, in commit
    self._prepare_impl()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 473, in _prepare_impl
    self.session.flush()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2459, in flush
    self._flush(objects)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2597, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2557, in _flush
    flush_context.execute()
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 586, in execute
    persistence.save_obj(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 239, in save_obj
    _emit_insert_statements(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 1137, in _emit_insert_statements
    result = cached_connections[connection].execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 287, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1101, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1247, in _execute_context
    self._handle_dbapi_exception(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1243, in _execute_context
    self.dialect.do_execute(
  File "/home/stefan/envs/py38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
```

**Comment by profjsb** [comment]
The phone number issue is addressed in #898

**Comment by dmitryduev** [comment]
We have cleaned up most of the stuff, so closing.
